code/clue.js

```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  suspects,
+  weapons,
+  rooms:[]
+}
 
+//Killer
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
+
   mystery.killer = randomSelector(suspects)
+  killerImage.src = mystery.killer.image
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
   document.getElementById(
     'killerName'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById(
+    'killerCard'
+    ).style.background = mystery.killer.color
+  }
+
+  document.getElementById(
+    'killerCard'
+    ).addEventListener('click', pickKiller);
+
+//Weapons
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById(
+    'weaponName'
+).innerHTML = `${mystery.weapon.name}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+document.getElementById(
+  'weaponCard'
+  ).addEventListener('click', pickWeapon);
+
+//Rooms
+ const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById(
+    'roomName'
+  ).innerHTML = `${mystery.room}`
+ }
+
+ document.getElementById(
+  'roomCard'
+  ).addEventListener('click', pickRoom);
+
+// Reveal the mystery
+const revealMystery = (mystery) => {
+  event.preventDefault();
```

Comment:
`event` is no variable or some sort of that. It's an argument passed to your function that you need to provide. In your code, you called it `mystery`, so you should write `mystery.preventDefault()`